### DISCUSSION TITLE
Go to first/last sibling

### DISCUSSION THREAD

**Discussion by fuelen** [nvim-neo-tree/neo-tree.nvim#220] [Q&A] [answered]
I'm trying to migrate from `nvim-tree` and I'm looking for mapping for `K` and `J` keys, which are mapped in `nvim-tree` to `first_sibling` and `last_sibling` actions respectively.

**Comment by cseickel** [comment]
I looked but couldn't find an explanation. What do those commands do exactly? 

Are they like what was discussed in these threads?
https://github.com/nvim-neo-tree/neo-tree.nvim/discussions/163
https://github.com/nvim-neo-tree/neo-tree.nvim/issues/201

**Reply by fuelen** [reply]
No, these commands weren't discussed there.

Let's say I have a folder with 5 files and my cursor is in the middle. `first_sibling` moves cursor to the first file and `last_sibling` moves cursor to the last file.

**Comment by cseickel** [comment, marked as answer]
Interesting. Here is how you can add those mappings:

```lua
  require('neo-tree').setup({
    window = {
      mappings = {
        ["J"] = function(state)
          local tree = state.tree
          local node = tree:get_node()
          local siblings = tree:get_nodes(node:get_parent_id())
          local renderer = require('neo-tree.ui.renderer')
          renderer.focus_node(state, siblings[#siblings]:get_id())
        end,
        ["K"] = function(state)
          local tree = state.tree
          local node = tree:get_node()
          local siblings = tree:get_nodes(node:get_parent_id())
          local renderer = require('neo-tree.ui.renderer')
          renderer.focus_node(state, siblings[1]:get_id())
        end
      }
    }
  })

```

**Reply by fuelen** [reply]
Thank you

**Reply by glaucocustodio** [reply]
Why not having these by default?

**Reply by cseickel** [reply]
I just generally prefer to make neo-tree easy to customize over having endless configuration options. At this point, I have all of the standard commands covered and would not add anymore unless there's something about that command which would make a custom command impossible or too complex.

In this case, it is a very simple command and would not benefit from being added to the built-in commands. If it were more popular and I got asked about it frequently, the next step would be to add it to the wiki.

**Reply by glaucocustodio** [reply]
ok, fair enough 🙂

I think would be useful linking this page on the README though as these functions are not that easy to implement in case someone need them.
